src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -122,6 +122,8 @@ public class LogManagerService extends PluginService {
     @Getter
     private int periodicUpdateIntervalSec;
     private Future<?> spaceManagementThread;
+    // public only for integ tests
+    public final AtomicBoolean isActiveFileCompleted = new AtomicBoolean(false);
```

Comment:
Exposing variables like this only for tests is not recommended. If you wanted to you could allow this service to have listeners that can get called when something happens. A very simple implementation for this:

```java
    private final List<Consumer<Event>> listeners = new ArrayList<>();

    enum Event {
          ACTIVE_FILE_COMPLETED
     }

     public registerlistener(Consumer<Event> listener) {
          listeners.add(listener);
     }

    private dispatchEvent(Event event) {
          for (Consumer<Event> listener : listeners) {
                listener(event);
           }
     }
```

even though for now it is only used by tests, listeners are opaque and do not know about the fact we store thing internally in a variable called `isActiveFileCompleted`. Additionally it allows you to do whatever you want and react to events.